### PR TITLE
Bug/140/is default

### PR DESCRIPTION
**Pull request by Jhalter5Stones**
set isDefault correctly looping through list of physical and a list of digital products as options in the sync config. set is default to true for one physical and one digital product

### CODE REVIEW

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Ready for review. Let me know if you see anything wrong with this

https://trello.com/c/V1BKzybj/140-dsbat-sync-products-at-a-click-of-a-button

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Can you take a look at this one more time before I send over to final review? Just to make sure we aren't missing anything

https://trello.com/c/V1BKzybj/140-dsbat-sync-products-at-a-click-of-a-button

File: lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb [hunk 1]
```diff
@@ -375,6 +375,30 @@ def process_products_and_variants(products, variants, links)
...
+          has_default = false
+          if product['model'].length == 1
+            product['model'][0]['isDefault'] = true
+            next
+          else
+            physical_formats.each do |val|
+              match = product['model'].select { |v| v[field_value(product, 'Inv_Product.Category')] == val }
+
+              if match
+                match['isDefault'] = true
```
Comment (mrogers-5s) [inline lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb:389]: You need the `break` after this line

File: lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb [hunk 2]
```diff
@@ -375,6 +375,30 @@ def process_products_and_variants(products, variants, links)
...
+              end
+            end
+
+            digital_formats.each do |val|
+              match = product['model'].select { |v| v[field_value(product, 'Inv_Product.Category')] == val }
+
+              if match
+                match['isDefault'] = true
```
Comment (mrogers-5s) [inline lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb:397]: You need the `break` after this line

File: lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb [hunk 3]
```diff
@@ -375,6 +375,30 @@ def process_products_and_variants(products, variants, links)
 
         result = []
         products_map.each_value { |p| result.push(p) }
+
+        result.each do |product|
+          has_default = false
```
Comment (mrogers-5s) [inline lib/huginn_acumen_product_agent/concerns/acumen_product_query_concern.rb:380]: You're not using this anywhere, so I think it can go away

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Changes made
@jspizziri I believe this is ready for final review and if this looks good to you when merged can you also push to ruby gems

https://trello.com/c/V1BKzybj/140-dsbat-sync-products-at-a-click-of-a-button

**Comment by jspizziri** [comment]
@Jhalter5Stones released as `v1.2.2`

**Pull request merged**